test/one/profiler/test/TestProcess.java

```diff
@@ -247,11 +247,24 @@ public Output profile(String args) throws IOException, TimeoutException, Interru
     }
 
     public Output profile(String args, boolean sudo) throws IOException, TimeoutException, InterruptedException {
+        return profile(args, sudo, "build/bin/asprof");
+    }
+
+    public Output profileWithCopy(String args) throws IOException, TimeoutException, InterruptedException {
+        return profileWithCopy(args, false);
+    }
+
+    // this profiles with a different copy of the profiler located in build_copy
+    public Output profileWithCopy(String args, boolean sudo) throws IOException, TimeoutException, InterruptedException {
```

Comment:
It's not necessary to use different `asprof`.
There is `--libpath` option where you can specify an alternate path to `libasyncProfiler.so`.